doc/changelog.md

```diff
@@ -1,55 +1,100 @@
-# Release 0.3.2-dev
+# Release 0.3.2
 
 <h3>New features</h3>
 
-<h3>Improvements</h3>
+* The experimental AutoGraph feature now supports Python `while` loops, allowing native Python loops
+  to be captured and compiled with Catalyst.
+  [(#318)](https://github.com/PennyLaneAI/catalyst/pull/318)
 
-* Improve the compiler driver diagnostic output even more. Now it dumps the failing IR as well as
-  prints names of failing passes.
-  [(#349)](https://github.com/PennyLaneAI/catalyst/pull/349)
+  ```python
+  dev = qml.device("lightning.qubit", wires=4)
 
-* Return values of conditional functions no longer need to be of exactly the same type. Catalyst
-  would apply type promotion to branch return values if their types don't match.
-  [(#333)](https://github.com/PennyLaneAI/catalyst/pull/333)
+  @qjit(autograph=True)
+  @qml.qnode(dev)
+  def circuit(n: int, x: float):
+      i = 0
+
+      while i < n:
+          qml.RX(x, wires=i)
+          i += 1
+
+      return qml.expval(qml.PauliZ(0))
+  ```
+
+  ```pycon
+  >>> circuit(4, 0.32)
+  array(0.94923542)
+  ```
+
+  This feature extends the existing AutoGraph support for Python `for` loops and if` statements
+  introduced in v0.3. Note that TensorFlow must be installed for AutoGraph support.
+
+  For more details, please see the
+  [AutoGraph guide](https://docs.pennylane.ai/projects/catalyst/en/stable/dev/autograph.html).
+
+* In addition to loops and conditional branches, AutoGraph now supports native Python `and`, `or`
+  and `not` operators in Boolean expressions.
+  [(#325)](https://github.com/PennyLaneAI/catalyst/pull/325)
 
   ```python
-  @qjit
-  def func(i: int, f: float):
-      @cond(i < 3)
-      def cond_fn():
-          return i
-      @cond_fn.otherwise
-      def otherwise():
-          return f
-      return cond_fn()
+  dev = qml.device("lightning.qubit", wires=1)
+
+  @qjit(autograph=True)
+  @qml.qnode(dev)
+  def circuit(x: float):
+
+      if x >= 0 and x < jnp.pi:
+          qml.RX(x, wires=0)
+
+      return qml.probs()
   ```
 
   ```pycon
-  >>> func(1, 4.0)
-  array(1.0)
+  >>> circuit(0.43)
+  array([0.95448287, 0.04551713])
+  >>> circuit(4.54)
+  array([1., 0.])
   ```
 
-* Improve the `CopyGlobalMemRefPass` of our MLIR processing pipeline by adding the support of
-  dynamically shaped arrays.
-  [(#348)](https://github.com/PennyLaneAI/catalyst/pull/348)
+  Note that Boolean expressions will only be captured by AutoGraph if at least
+  one of the arguments is a numeric object.
 
-* Improve the compiler driver diagnostic output. The driver now provides more context for error
-  messages and includes a verbose trace if verbose mode is enabled.
-  [(#303)](https://github.com/PennyLaneAI/catalyst/pull/303)
+* It is now possible to dynamically load third-party Catalyst compatible devices directly
+  into a pre-installed Catalyst runtime.
```

Comment:
It turned out we need to do more work to get this working on macOS :( 

```suggestion
* It is now possible to dynamically load third-party Catalyst compatible devices directly
  into a pre-installed Catalyst runtime on Linux.
```

Reply:
oh really? Not MacOS?

Reply:
No, there are some API incompatibilities :(